pkgs/applications/editors/bluej-bin/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, makeWrapper, unzip, jetbrains
+}:
+let
+  version = "414";
+in
+stdenv.mkDerivation {
+  name = "bluej-bin-${version}";
+  src = fetchurl {
+    url = "https://www.bluej.org/download/files/BlueJ-linux-${version}.deb";
+    sha512 = "c9c5c905d6dcf4388fe1423179be836ad8d35a4d197e53706a5764d9192c439aa60b80a28b87222b9f16ed1534d7a7768d924fc06ddd95107aea3cb00570c22a";
+  };
+
+  buildCommand = ''
```

Comment:
I had a chance to play with this a little bit and this might not be prefect but something like this would be preferable:

```
  dontBuild = true;
  propagatedBuildInputs = [ jetbrains.jdk ];
  nativeBuildInputs = [ makeWrapper ];

  unpackPhase = ''
    ar xf $src
    tar xf data.tar.xz
  '';

  installPhase = ''
    mkdir -p $out
    cp -r usr/* $out/

    makeWrapper ${jetbrains.jdk}/bin/java $out/bin/bluej \
      --add-flags "-Djavafx.embed.singleThread=true -Dawt.useSystemAAFontSettings=on -Xmx512M -cp \"$out/share/bluej/bluej.jar:${jetbrains.jdk}/lib/tools.jar\" bluej.Boot"
  '';
```